shell/src/main/java/alluxio/shell/command/CopyFromLocalCommand.java

```diff
@@ -80,15 +79,75 @@ public void run(CommandLine cl) throws IOException {
   }
 
   /**
+   * Copies a directory from local to Alluxio filesystem. The destination directory structure
+   * maintained as local directory. This method is used when input path is a directory.
+   *
+   * @param srcDir The source directory in the local filesystem
+   * @param dstPath The {@link AlluxioURI} of the destination
+   * @throws IOException if a non-Alluxio related exception occurs
+   *
+   */
+  private void copyFromLocalDirs(File srcDir, AlluxioURI dstPath) throws IOException {
+    try {
+      createDscDir(dstPath);
+      List<String> errorMessages = Lists.newArrayList();
+      File[] fileList = srcDir.listFiles();
+      for (File srcFile : fileList) {
+        try {
+          if (srcFile.isDirectory()) {
+            copyFromLocalDirs(srcFile, new AlluxioURI(dstPath, new AlluxioURI(srcFile.getName())));
+          } else {
+            copyPath(srcFile, new AlluxioURI(dstPath, new AlluxioURI(srcFile.getName())));
+          }
+        } catch (IOException e) {
+          errorMessages.add(e.getMessage());
+        }
+      }
+      if (errorMessages.size() != 0) {
+        if (errorMessages.size() == fileList.length) {
```

Comment:
I feel this check is not sufficient to tell no file is created: it's possible that some files are created in the subdirectory from the call `copyFromLocalDirs` and then failed there, right?